src/qtt/algorithms/functions.py

```diff
@@ -215,13 +215,49 @@ def _double_gaussian(x, A_dn, A_up, sigma_dn, sigma_up, mean_dn, mean_up):
     result_dict = {'parameters initial guess': initial_params,
                    'separation': separation, 'split': weigthed_distance_split}
     result_dict['parameters'] = par_fit
+    result_dict['reduced_chi_squared'] = result.redchi
+
     result_dict['left'] = np.take(par_fit, [4, 2, 0])
     result_dict['right'] = np.take(par_fit, [5, 3, 1])
     result_dict['type'] = 'fitted double gaussian'
 
     return par_fit, result_dict
 
 
+def refit_double_gaussian(result_dict, x_data, y_data, gaussian_amplitude_ratio_threshold=8):
+    """ Improve fit of double Gaussian by estimating the initial parameters based on an existing fit
```

Comment:
I know broadly what is happening in this method but I find it hard to conclude that all the mathematics is correct.

Reply:
Any suggestions on how to improve? I split part of the function and the other suggestions also helped to clear up the code.